invisible_cities/detector_simulation/buffer_functions.py

```diff
@@ -0,0 +1,143 @@
+                      pmt_binwid: float, sipm_binwid: float) -> Callable:
+    """
+    Calculates the output buffers for all sensors
+    based on a configured buffer length and pretrigger.
+    Synchronising the clock between the two sensor types.
+
+    Parameters
+    ----------
+    buffer_len  : float
+                  Length of buffer expected in mus
+    pre_trigger : float
+                  Time in buffer before identified signal in mus
+    pmt_binwid  : float
+                  Width in mus of PMT sample integration
+    sipm_binwid : float
+                  Width in mus of SiPM sample integration
+    """
+    pmt_buffer_samples  = int(buffer_len /  pmt_binwid)
+    sipm_buffer_samples = int(buffer_len / sipm_binwid)
+    sipm_pretrg         = int(pre_trigger / sipm_binwid)
+    sipm_postrg         = sipm_buffer_samples - sipm_pretrg
+    pmt_pretrg_base     = int(pre_trigger / pmt_binwid)
+    pmt_postrg_base     = pmt_buffer_samples - pmt_pretrg_base
+
+    def generate_slice(trigger    : int       ,
+                       pmt_bins   : np.ndarray,
+                       pmt_charge : np.ndarray,
+                       sipm_bins  : np.ndarray,
+                       sipm_charge: np.ndarray) -> Tuple:
+        """
+        Synchronises the clocks between the SiPMs and PMTs
+        and slices the histograms where this synchronisation
+        indicates.
+        """
+        npmt_bin  = len(pmt_bins)
+        nsipm_bin = len(sipm_bins)
+
+        trg_bin    = np.where(sipm_bins <= pmt_bins[trigger])[0][-1]
+        bin_corr   = (pmt_bins[trigger] - sipm_bins[trg_bin]) // pmt_binwid
+        pmt_pretrg = pmt_pretrg_base + int(bin_corr)
+        pmt_postrg = pmt_postrg_base - int(bin_corr)
+
+        pmt_pre    = 0       , trigger - pmt_pretrg
+        pmt_pos    = npmt_bin, trigger + pmt_postrg
+        pmt_slice  = slice(max(pmt_pre), min(pmt_pos))
+        pmt_pad    = -min(pmt_pre), max(0, pmt_pos[1] - npmt_bin)
+
+        sipm_pre   = 0        , trg_bin - sipm_pretrg
+        sipm_pos   = nsipm_bin, trg_bin + sipm_postrg
+        sipm_slice = slice(max(sipm_pre), min(sipm_pos))
+        sipm_pad   = -min(sipm_pre), max(0, sipm_pos[1] - nsipm_bin)
+
+        return (pad_safe( pmt_charge[:,  pmt_slice],  pmt_pad),
+                pad_safe(sipm_charge[:, sipm_slice], sipm_pad))
+
+
+    def position_signal(triggers   :       List,
+                        pmt_bins   : np.ndarray,
+                        pmt_charge :  pd.Series,
+                        sipm_bins  : np.ndarray,
+                        sipm_charge:  pd.Series
+                        ) -> List[Tuple[np.ndarray, np.ndarray]]:
+        """
+        Synchronises the SiPMs and PMTs for each identified
+        trigger and calls the padding function to fill with
+        zeros where necessary.
+        """
+        pmt_q  = np.array(pmt_charge.tolist())
+        sipm_q = np.empty((0,0))\
+          if sipm_charge.empty else np.array(sipm_charge.tolist())
```

Comment:
I think this would do what you want, if you want to give it a try:
```suggestion
        sipm_q = np.asarray(sipm_charge.tolist(), ndmin=2)
```